### PR TITLE
feat(contract) initial POC of poap entitlement custom

### PR DESCRIPTION
**Pull request by giuseppecrj**

### CODE REVIEW

File: contracts/src/spaces/entitlements/poap/PoapEntitlement.sol [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    for (uint256 i = 0; i < users.length; i++) {
+      if (_hasEventPoap(users[i], eventId)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  function schema() public pure returns (string memory) {
+    return "uint256 eventId";
```
Comment (clemire) [inline contracts/src/spaces/entitlements/poap/PoapEntitlement.sol:32]: For a parameter list with multiple params, would this be comma-separated?
e.g.

"uint256 eventId, address someContractAddress"

**Review by clemire — COMMENTED** [review]
comment inline

File: contracts/src/spaces/entitlements/poap/PoapEntitlement.sol (hunk 1 above)
Comment (giuseppecrj) [reply contracts/src/spaces/entitlements/poap/PoapEntitlement.sol:32]: right, here's an example: https://sepolia.easscan.org/schema/view/0x4f532d4a635375689656f9fe828c8baaa7d475c90e2dbcd381580d65598244f8

File: contracts/src/spaces/entitlements/poap/PoapEntitlement.sol [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    for (uint256 i = 0; i < users.length; i++) {
+      if (_hasEventPoap(users[i], eventId)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  function schema() public pure returns (string memory) {
```
Comment (brianathere) [inline contracts/src/spaces/entitlements/poap/PoapEntitlement.sol:31]: suggest getParamsSchema

File: contracts/src/spaces/entitlements/poap/PoapEntitlement.sol [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+contract PoapEntitlement {
+  IPOAP immutable poapContract;
+
+  constructor(address poapContractAddress) {
+    poapContract = IPOAP(poapContractAddress);
+  }
+
+  function isEntitled(
+    address[] calldata users,
+    bytes calldata data
```
Comment (brianathere) [inline contracts/src/spaces/entitlements/poap/PoapEntitlement.sol:19]: aligned with getParamsSchema, I'd name this params

File: contracts/src/spaces/entitlements/poap/PoapEntitlement.sol (hunk 1 above)
Comment (brianathere) [reply contracts/src/spaces/entitlements/poap/PoapEntitlement.sol:32]: Is this an EIP based way of doing this? It's basically just using method selector encoding while retaining the variable name. If it's a standard, then I guess we can start with this, though I'd like to also have a description. Let's hop on a call.

**Comment by giuseppecrj** [comment]
Updated with new changes @clemire

File: contracts/src/spaces/entitlements/ICrossChainEntitlement.sol [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    string primitive;
+    string description;
+  }
+
+  function isEntitled(
+    address[] calldata users,
+    bytes calldata data
+  ) external view returns (bool);
+
+  function properties() external view returns (Property[] memory);
```
Comment (brianathere) [inline contracts/src/spaces/entitlements/ICrossChainEntitlement.sol:21]: I believe parameters is more appropriate, as this is describing parameters that must be passed, encoded in this way, as parameters to the isEntitled call, and not actual properties of the ICrossChainEntitlement as the name properties implies.

**Review by brianathere — COMMENTED** [review]
properties -> parameters

File: contracts/src/spaces/entitlements/ICrossChainEntitlement.sol (hunk 4 above)
Comment (giuseppecrj) [reply contracts/src/spaces/entitlements/ICrossChainEntitlement.sol:21]: sounds good, short term memory thought you said properties and not params during our cal. Will fix soon!

**Pull request merged**